packages/grid/x-data-grid-pro/src/hooks/features/infiniteLoader/useGridInfiniteLoader.ts

```diff
@@ -27,57 +25,63 @@ export const useGridInfiniteLoader = (
 ): void => {
   const visibleColumns = useGridSelector(apiRef, gridVisibleColumnDefinitionsSelector);
   const currentPage = useGridVisibleRows(apiRef, props);
-  const rowsMeta = useGridSelector(apiRef, gridRowsMetaSelector);
-  const contentHeight = Math.max(rowsMeta.currentPageTotalHeight, 1);
+  const observer = React.useRef<IntersectionObserver>();
+  let previousY = 0;
+  let previousRatio = 0;
 
-  const isInScrollBottomArea = React.useRef<boolean>(false);
+  const handleLoadMoreRows = React.useCallback(
```

Comment:
In some cases the callback is fired twice, see this demo: https://codesandbox.io/s/sweet-wing-y3h4zt?file=/demo.tsx

Expected behavior: on scroll, additional 3 rows are fetched which results in total of 9 rows
Current behavior: on scroll, 2x 3 rows are fetched which results in total of 12 rows

https://github.com/mui/mui-x/assets/13808724/4d91bab0-a3c9-4932-906a-9086ec14df16

Reply:
I tried to debug it but as far as I can see this happens either because of the accelerated scroll on IOS or another option is because you are trying to load fewer rows that the screen can fit, essentially hitting the intersection again. The callback passes the `viewportPageSize` to indicate how many rows the user can load.

Reply:
I think I can reproduce the double fetching with our demo in https://deploy-preview-8672--material-ui-x.netlify.app/x/react-data-grid/row-updates/#infinite-loading
You can see this if you add logging: https://codesandbox.io/s/compassionate-panka-flyq4j?file=/demo.tsx

https://github.com/mui/mui-x/assets/13808724/2f5fb8a8-ec05-4801-995b-d03c82b51057

You don't see a visual reproduction of the issue, because the demo has a hidden bug where the second response overrides data fetched in the first one:
https://github.com/mui/mui-x/blob/be7174996bc1157dd1d39343c819b6a9af7581bb/docs/data/data-grid/row-updates/InfiniteLoadingGrid.tsx#L38

I think this demo is generally buggy and needs a rewrite (I'll try to create a better one).
But the double fetching issue is consistent and needs a solution.

Reply:
@DanailH